fabric-brewing-v1/src/main/java/net/fabricmc/fabric/impl/brewing/PotionIngredient.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.brewing;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionUtil;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.PacketByteBuf;
+import net.minecraft.util.registry.Registry;
+
+public class PotionIngredient {
```

Comment:
Should be a custom ingredient entry instead of a new class which no one knows if this is actually backed by a potion+type or a vanilla ingredient

Reply:
The `Ingredient` class is unfortunately `final`, or I would have extended it directly as I'd much prefer to. How would you suggest implementing this better while still fully supporting `Ingredient` as a valid value?

Reply:
Ah, keep this one. My bad! Thought Ingredient Entry was public. Time for me to consider exposing it!

Reply:
Ah, that's what you meant. `Ingredient.Entry` is public, but there isn't a way to override the methods on `Ingredient` for parsing/creating `Entry`s without a mixin, so it would be quite the process to wrap in that way. I'll see if it's feasible, though.

Reply:
I suggest leaving that for this pr. Another one that correctly fixes this issue can come back to this later and serialize things properly instead.